src/ExpressReceiver.ts

```diff
@@ -96,12 +96,24 @@ export default class ExpressReceiver implements Receiver {
 
     try {
       await this.bolt?.processEvent(event);
-      if (storedResponse !== undefined) {
-        if (typeof storedResponse === 'string') {
-          res.send(storedResponse);
-        } else {
-          res.json(storedResponse);
+
+      if (this.processBeforeResponse) {
+        let spentMillis = 0;
+        const millisForTimeout = 10;
+        // Wait here until the isAcknowledged is marked as true or 3 seconds have passed
```

Comment:
I know this is not ideal but I haven't figured out any other solutions yet.

Reply:
I'm a little confused by this. Can you explain the error use case you are fixing here?

Reply:
@stevengill 
I didn't mention clearly in the description (sorry!) but I have to say that `processBeforeResponse` just doesn't work for most cases (I haven't verified if it actually works in the code review, so that was not able to detect it 🙇 ). There are two issues with the current implementation as below:
* `processBeforeResponse` doesn't respond to non-event requests such as commands (listeners are executed but no HTTP response is submitted - resulting in timeouts)
* For Events API, `processBeforeResponse` doesn't wait for the completion of listeners and responds to incoming requests immediately (= it's not working as the solution for FaaS compatibility)

The change here is addressing the latter one.

This fix surely holds off responding before verifying the acknowledgment on the `App`'s processEvent method side. It works. However, I'm a bit hesitant to go with this "sleep" approach in Node.js. The most mysterious behavior I haven't fully understood yet is that `await this.bolt?.processEvent(event);` doesn't wait for the completion of listeners. The line returns before the completion of `const settledListenerResults = await allSettled(listenerResults)` in the method. That's why I had to implement this part in the `Receiver` side. I'm still trying to get rid of this part. It'd be greatly appreciated if you could dive deep into the issue.

Reply:
One additional note is that I've added a unit test verifying `App`'s behavior with `processBeforeResponse`. But I haven't added any tests for `ExpressReceiver` for it. This means even all the unit tests have passed, it doesn't mean the behavior is valid. It's necessary to make sure if it works in real apps. I was going to add tests to the `ExpressReceiver` side as well but if we could get rid of this "sleep" part, probably it is no longer needed.